src/toil/test/src/jobTestCacheEjection.py

```diff
@@ -0,0 +1,87 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+import random
+import os
+
+from toil.job import Job
+from toil.test import ToilTest
+
+class JobCacheEjectionTest(ToilTest):
+    """
+    Tests testing the Job.FileStore class
+    """
+    def testCacheEjection(self):
+        """
+        Makes three jobs that create an output file each which they write to filestore.  The combined size of any two
+        files is always less that cacheSize but the combined size of all 3 is always more so 1 file always has to be
+        ejected. Test to ensure that A is always ejected regardless of size.
+        """
+        for test in xrange(10):
+            options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+            options.logLevel = "INFO"
+            options.cacheSize = 100000
+            options.retryCount=100
+            options.badWorker=0.5
+            options.badWorkerFailInterval = 1.0
+            options.logFile = 'toil_CacheEject_log.txt'
+            file_sizes = [50000, 40000, 30000]
+            random.shuffle(file_sizes)
+            # Run the workflow, the return value being the number of failed jobs
+
+            A = Job.wrapJobFn(fileTestJob, file_sizes[0])
+            B = Job.wrapJobFn(fileTestJob, file_sizes[0])
+            C = Job.wrapJobFn(fileTestJob, file_sizes[0])
+            D = Job.wrapJobFn(fileTestCache, A.rv(), B.rv(), C.rv())
+            A.addChild(B)
+            B.addChild(C)
+            C.addChild(D)
+            Job.Runner.startToil(A, options)
+
+            job_passed=True
+            #with open(options.logFile, 'r') as outfile:
+            #    for line in outfile:
+            #        if 'failed with exit value -9' in line:
+            #            job_passed=False
+            #            break
+            #os.remove(options.logFile)
+            assert job_passed
+
+
+
+def fileTestCache(job, file_a, file_b, file_c):
+    """
+    Test job exercises Job.FileStore functions
+    """
+    assert file_a not in job.fileStore.jobStoreFileIDToCacheLocation.items()
+
+def fileTestJob(job, file_size):
+    """
+    Test job exercises Job.FileStore functions
+    """
+    def randomString(str_len):
+        chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        s = "".join(map(lambda i : random.choice(chars), xrange(str_len)))
+        return s
+    if random.random() > 0.5:
+        #Make a local copy of the file
+        tempFile = job.fileStore.getLocalTempFile()
+    else:
+        tempDir = job.fileStore.getLocalTempDir()
+        tempFile = ''.join([tempDir, '/', randomString(8), '.txt'])
+    with open(tempFile, 'w') as fH:
+        fH.write(randomString(file_size))
```

Comment:
will do. @benedictpaten  I got the randomString def from jobFileStoreTest. Maybe that should incorporate os.urandom too?